docs/security/security-best-practices.md

```diff
@@ -4,37 +4,67 @@ menu_title: Security best practices
+Security is extremely important for online stores. A breach can lead to significant financial loss, reputational damage, and erosion of customer trust. Implementing robust security practices is crucial. This guide covers the best practices for securing WooCommerce stores, including hardening WordPress, keeping plugins and themes up to date, implementing secure coding practices, and protecting user data. By following these recommendations, developers can build secure and resilient WooCommerce stores that protect both their business and their customers.
 
-## Audience
+### Audience
 
 This guide is intended for developers who are familiar with WordPress and WooCommerce and want to improve the security of their online stores.
 
-## Prerequisites
+### Prerequisites
 
 To follow this guide, you should have:
 
 1. A basic understanding of WordPress and WooCommerce.
 2. Access to a WordPress website with WooCommerce installed and activated.
 
-## Step 1 - Keep WordPress, WooCommerce, and plugins up to date
+## Security Standards
+
+We recommend that you maintain the following standards: 
+
+- **PHPCS**: The PHP Code Sniffer tool helps ensure your code adheres to specific coding standards, which can prevent common security issues. Extensions have no security issues when testing against PHPCS rules.
+
+- **SemGrep**: Focuses on identifying patterns that may indicate security vulnerabilities, allowing developers to write more secure code. Extensions have no security issues when testing against SemGrep rules.
+
+Understanding and implementing these standards can significantly reduce the risk of security vulnerabilities in your extensions and themes.
+
+### Common Security Pitfalls and How to Avoid Them
+
+Several common security pitfalls can easily be avoided with vigilance and best practices:
+
+- **SQL Injection**: Ensure all database queries are properly sanitized.
+- **Cross-Site Scripting**: Sanitize all user input to prevent malicious code from being executed.
+- **File Inclusions**: Limit file inclusion to prevent unauthorized access to the file system.
+
+Awareness and prevention of these common issues are crucial steps in securing your WooCommerce store.
+
+## Manual Testing Guidelines
+
+Even without access to automated tools like QIT, manual testing remains a vital component of securing your WooCommerce store. Here are some guidelines on how developers can manually test their extensions for security vulnerabilities:
+
+1. **Code Review**: Regularly review your code for security vulnerabilities, focusing on the areas where breaches commonly occur, such as user input fields and database queries.
+2. **Use Security Plugins**: Plugins like Wordfence, Sucuri, or iThemes Security provide an interface for manual security checks, including file integrity monitoring and malware scanning.
+3. **Testing Environments**: Utilize staging environments to test the security of new features or updates before deploying them live.
```

Comment:
We could also suggest tools like OWASP Zap etc here as well. What do you think?

Reply:
@zhongruige I'm not very familiar with that tool. I might need a suggestion for this change, or I can ask others from my team to weigh in.

Reply:
Oh no worries @brentmackinnon! The tool mentioned is this one: https://www.zaproxy.org/

Primarily it's to just note that possibly we could switch the wording on number two to be something like:


2.  **Use Security Plugins and Tooling**: Plugins like Wordfence, Sucuri, or iThemes Security provide an interface for manual security checks, including file integrity monitoring and malware scanning. Additionally, tooling like [OWASP Zap](https://www.zaproxy.org/) or [GitHub Security Scanning](https://github.com/features/security/) can be used.